src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2185,14 +2189,21 @@ namespace winrt::TerminalApp::implementation
             {
                 co_await winrt::resume_foreground(Dispatcher());
 
+                if (warnMultiLine)
+                {
+                    const auto focusedTab = _GetFocusedTabImpl();
```

Comment:
This is awkward. I wish I can do this in the background thread. But If I do that, it just *crashes*. So I have to do it here. And the logic is kinda messed up.

Reply:
Probably due to the call to 
```c++
        if (_tabView.TabItems().IndexOf(_tabView.SelectedItem(), focusedIndex))
```

Presumably that has to happen on the FG thread

Reply:
My first attempt is to do the check outside `if (warnMultiLine || warnLargeText)`, which is much more streamlined. But it'll crash. So I have to do this inside `if (warnMultiLine || warnLargeText)` and double-check it after switching to FG thread.